lib/src/authentication/src/auth.dart

```diff
@@ -63,7 +63,7 @@ abstract class Auth {
   /// with, the default values. To understand why an Ably [TokenRequest] may be
   /// issued to clients in favor of a token, see
   /// [Token Authentication explained](https://ably.com/docs/core-features/authentication/#token-authentication).
-  Future<TokenDetails> requestToken({
+  Future<TokenDetails?> requestToken({
```

Comment:
What's the reason for making these optional? According to the spec these methods should never return nil

Reply:
Good point. I suppose I was trying to stay in line with the return type of `invoke` which was optional. But looks like there is `invokeRequest` that does have a non-null return value. Changes here 8300e231bb8244d85c72e5d2be6ca47b34df61b5